src/test/java/tutoraid/model/student/ProgressListTest.java

```diff
@@ -0,0 +1,174 @@
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new Progress(null));
+    }
+
+    @Test
+    public void isValidProgressList() {
+        // null progress list
+        Assert.assertThrows(NullPointerException.class, () -> ProgressList.isValidProgressList(null));
+
+        // invalid progress list
+        ArrayList<String> moreThanTenElements = new ArrayList<>();
+        for (int i = 0; i < 11; i++) { // too many elements
+            moreThanTenElements.add("abc"); // valid progress description
+        }
+        assertFalse(ProgressList.isValidProgressList(moreThanTenElements));
+
+        ArrayList<String> nullElements = new ArrayList<>();
+        nullElements.add(null);
+        assertFalse(ProgressList.isValidProgressList(nullElements));
+
+        ArrayList<String> invalidProgress = new ArrayList<>();
+        invalidProgress.add("abc"); // valid progress description
+        invalidProgress.add("exam*"); // invalid progress description
+        assertFalse(ProgressList.isValidProgressList(invalidProgress));
+
+        // valid progress list
+        ArrayList<String> lessThanTenElements = new ArrayList<>();
+        for (int i = 0; i < 9; i++) { // less than 10 elements
+            moreThanTenElements.add("abc"); // valid progress description
+        }
+        assertTrue(ProgressList.isValidProgressList(lessThanTenElements));
+
+        ArrayList<String> tenElements = new ArrayList<>();
+        for (int i = 0; i < 10; i++) { // 10 elements
+            moreThanTenElements.add("abc"); // valid progress description
+        }
+        assertTrue(ProgressList.isValidProgressList(tenElements));
+    }
+
+    @Test
+    public void addProgress_nullProgress_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> progressList.addProgress(null));
+    }
+
+    @Test
+    public void addProgress_validProgress_addsNewProgress() {
+        ArrayList<String> updatedProgressListDescriptions = new ArrayList<>();
+        ProgressList updatedProgressList = new ProgressList();
+
+        for (int i = 0; i < 10; i++) {
+            updatedProgressListDescriptions.add(String.valueOf(i));
+            updatedProgressList.addProgress(new Progress(String.valueOf(i)));
+        }
+
+        assertTrue(updatedProgressList.equals(new ProgressList(updatedProgressListDescriptions)));
+    }
+
+    @Test
+    public void addProgress_fullList_addsNewProgressAndReplaceTheLatestOne() {
+        ArrayList<String> oldProgressListDescriptions = new ArrayList<>();
+        ArrayList<String> updatedProgressListDescriptions = new ArrayList<>();
+
+        for (int i = 0; i < 10; i++) {
+            oldProgressListDescriptions.add(String.valueOf(i));
+            updatedProgressListDescriptions.add(String.valueOf(i + 1));
+        }
+
+        ProgressList amyOldProgressList = new ProgressList(oldProgressListDescriptions);
```

Comment:
I wanted to check what `amyOldProgressList` and `amyUpdatedProgressList` mean? Are they maybe spelling errors?

Reply:
What I meant is Amy's old progress list and updated progress list. Maybe I will remove "amy" since it might be confusing.